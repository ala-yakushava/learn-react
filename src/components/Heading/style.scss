@import "../../styles/theme";

.Heading {
  margin-bottom: 40px;
  font-size: 40px;
  text-transform: uppercase;
};
