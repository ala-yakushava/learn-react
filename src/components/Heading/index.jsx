import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './style.scss';

const Heading = ({ children, className }) => {
  const headingClass = cn({
    Heading: true,
    [className]: true,
  });

  return (
    <div className={headingClass}>
      { children }
    </div>
  );
};

Heading.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

Heading.defaultProps = {
  children: null,
  className: '',
};

export default Heading;
