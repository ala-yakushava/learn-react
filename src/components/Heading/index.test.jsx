import React from 'react';
import renderer from 'react-test-renderer';

import Heading from './index';

describe('Components - Heading', () => {
  test('should match snapshot', () => {
    const tree = renderer
      .create(<Heading />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
