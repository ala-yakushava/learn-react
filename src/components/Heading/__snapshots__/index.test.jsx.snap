// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Heading should match snapshot 1`] = `
<div
  className="Heading"
/>
`;
